packages/sdk/src/react/_internal/transaction-machine/execute-transaction.ts

```diff
@@ -157,13 +147,46 @@ export class TransactionMachine {
 		private readonly openSelectPaymentModal: (
 			settings: SelectPaymentSettings,
 		) => void,
+		private readonly accountChainId: number,
```

Comment:
This can not be a primitive value here. the chain can change after the class is created. Ether trough the switch chain, or by the user in his wallet.
Previously we took it from the account, so its always up to date